aries_cloudagent/wallet/routes.py

```diff
@@ -757,6 +786,90 @@ async def wallet_set_did_endpoint(request: web.BaseRequest):
         return web.json_response({"txn": transaction.serialize()})
 
 
+@docs(tags=["wallet"], summary="Create a EdDSA jws using did keys with a given payload")
+@request_schema(JWSCreateSchema)
+@response_schema(WalletModuleResponseSchema(), description="")
+async def wallet_jwt_sign(request: web.BaseRequest):
+    """
+        Request handler for jws creation using did.
+
+    Args:
+        "headers": { ... },
+        "payload": { ... },
+        "did": "did:example:123",
+        "verificationMethod": "did:example:123#keys-1"
+        with did and verification being mutually exclusive.
+    """
+    context: AdminRequestContext = request["context"]
+    body = await request.json()
+    did = body.get("did")
+    if did is None:
+        did = body.get("verificationMethod").split("#", 1)[0]
+    if did is None:
+        raise web.HTTPBadRequest(reason="did and verificationMethod required.")
+    payload = body["payload"]
+    headers = body.get("headers", {})
+    async with context.session() as session:
+        wallet = session.inject_or(BaseWallet)
+        if not wallet:
+            raise web.HTTPForbidden(reason="No wallet available")
+        try:
+            did_info = await wallet.get_local_did(did)
+            headers["alg"] = "EdDSA"
+            headers["typ"] = "JWT"  # TODO: why not JWS?
+            headers["kid"] = f"{did}#keys-1"  # TODO: fix me
+            headers["crit"] = ["b64"]  # TODO: do we need this
+            # TODO: do we need to b64encode before signing?
+            signature = await wallet.sign_message(
+                f"{headers.encode('utf-8')}.{payload.encode('utf-8')}", did_info.verkey
+            )
+        except WalletNotFoundError as err:
+            raise web.HTTPNotFound(reason=err.roll_up) from err
+        except WalletError as err:
+            raise web.HTTPBadRequest(reason=err.roll_up) from err
+    sig = bytes_to_b64(signature, pad=False)
+    return web.json_response(f"{headers}.{payload}.{sig}")
+
+
+@docs(tags=["wallet"], summary="Verify a EdDSA jws using did keys with a given JWS")
+@request_schema(JWSVerifySchema)
+@response_schema(WalletModuleResponseSchema(), description="")
+async def wallet_jwt_verify(request: web.BaseRequest):
+    """
+        Request handler for jws validation using did.
+
+    Args:
+        "jwt": { ... }
+    """
+    context: AdminRequestContext = request["context"]
+    body = await request.json()
+    jwt = body["jwt"]
+    encoded_header, encoded_payload, encoded_signiture = jwt.split(".", 3)
+    header = json.loads(b64decode(encoded_header))
+
+    if (
+        "alg" not in header
+        or header["alg"] != "EdDSA"
+        or "b64" not in header
+        or header["b64"] is not False
+        or "crit" not in header
+        or header["crit"] != ["b64"]
+        or "kid" not in header
+    ):
+        raise BadJWSHeaderError(
+            "Invalid JWS header parameters for Ed25519Signature2018."
+        )
```

Comment:
I get the itch that there is a more elegant way to do this but everything coming to mind doesn't seem any simpler lol. I think the `b64` header should not be required, as discussed in a previous comment.